lib/pipeline.js

```diff
@@ -197,10 +197,11 @@ Pipeline.prototype.execBuffer = util.deprecate(function () {
   return execBuffer.apply(this, arguments);
 }, 'Pipeline#execBuffer: Use Pipeline#exec instead');
 
+var exec = Pipeline.prototype.exec;
 Pipeline.prototype.exec = function (callback) {
   if (this._transactions > 0) {
     this._transactions -= 1;
-    return execBuffer.apply(this, arguments);
+    return exec.apply(this, arguments);
```

Comment:
`return (this.options.dropBufferSupport ? exec : execBuffer).apply(this, arguments);` should fix the tests in "test/functional/transaction.js"